tests/provisioning/loadgen_test.py

```diff
@@ -52,7 +52,7 @@ def testNumberOfNode(self):
 
     def testReachOfLoadgen(self):
         """Test if querying load generator returns 200"""
-        command = ("kubectl get service locust-main --context=loadgenerator -o jsonpath='{.status.loadBalancer.ingress[0].ip}'")
+        command = ("kubectl get service locust-main -o jsonpath='{.status.loadBalancer.ingress[0].ip}'")
```

Comment:
why did you remove the context? What will happen in case when two test runs happen on the same worker in parallel such that both use kubernetes clusters but one expects loadgen k8s and another expects hipstershop k8s?

Reply:
The context was causing the tests to break - I assume the context name wasn't set as expected in the container?

These tests are made to run inside a container, so I don't think we have to worry about context overlap as long as we follow the pattern of restricting each container to a single main process. The context is set at the top of each test as part of the setup function

Reply:
The tests should not be aware of the environment in which they are running. So even though we run tests in a container, we should still be careful when we use shared resources. 
So if we know the context name, we should update line 55 to include the right name.
If we don't know or it might take time to troubleshoot it and we want to unblock the tests, we can go ahead with the current fix but also adding a TODO comment explaining what's problematic here so we can fix it in the future.